apps/cowswap-frontend/src/modules/tradeQuote/hooks/useTradeQuoteManager.ts

```diff
@@ -0,0 +1,62 @@
+import { useSetAtom } from 'jotai/index'
+import { useMemo } from 'react'
+
+import { OrderQuoteResponse, PriceQuality } from '@cowprotocol/cow-sdk'
+
+import QuoteApiError, { QuoteApiErrorCodes } from 'api/cowProtocol/errors/QuoteError'
+import { FeeQuoteParams } from 'common/types'
+
+import { useProcessUnsupportedTokenError } from './useProcessUnsupportedTokenError'
+
+import { updateTradeQuoteAtom } from '../state/tradeQuoteAtom'
+import { SellTokenAddress } from '../state/tradeQuoteInputAtom'
+
+export interface TradeQuoteManager {
+  setLoading(hasParamsChanged: boolean): void
+  reset(): void
+  onError(error: QuoteApiError, requestParams: FeeQuoteParams): void
+  onResponse(data: OrderQuoteResponse, requestParams: FeeQuoteParams, fetchStartTimestamp: number): void
+}
+
+export function useTradeQuoteManager(sellTokenAddress: SellTokenAddress | undefined): TradeQuoteManager | null {
```

Comment:
Extracted the `tradeQuotesAtom` state updates logic in `TradeQuoteManager` in order to make it more clear what are the updates doing.